tools/cluster/tests/cluster_stability_test.go

```diff
@@ -0,0 +1,574 @@
+	// Give the test time to start
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		e.chainEnv.t.Log("Sabotaging the following nodes:\n")
+		e.chainEnv.t.Log(e.SabotageList)
+
+		time.Sleep(startDelay)
+
+		for _, nodeID := range e.SabotageList {
+			e.chainEnv.t.Logf("Breaking node: %v (%s)", nodeID, time.Now())
+
+			var err error
+
+			if sabotageOption == SabotageByKill {
+				err = e.chainEnv.clu.KillNode(nodeID)
+			} else if sabotageOption == SabotageByFreeze {
+				err = e.chainEnv.clu.FreezeNode(nodeID)
+			}
+
+			require.NoError(e.chainEnv.t, err)
+
+			time.Sleep(inBetweenDelay)
+		}
+
+		wg.Done()
+	}()
+
+	return &wg
+}
+
+func (e *SabotageEnv) unfreezeNodes() {
+	for _, nodeID := range e.SabotageList {
+		e.chainEnv.t.Logf("Unfreezing node %v", nodeID)
+		err := e.chainEnv.clu.UnfreezeNode(nodeID)
+
+		require.NoError(e.chainEnv.t, err)
+	}
+}
+
+func (e *SabotageEnv) getActiveNodeList() []int {
+	contains := func(x int) bool {
+		for _, n := range e.SabotageList {
+			if n == x {
+				return true
+			}
+		}
+
+		return false
+	}
+
+	activeNodeList := []int{}
+
+	for _, n := range e.chainEnv.clu.Config.AllNodes() {
+		if !contains(n) {
+			activeNodeList = append(activeNodeList, n)
+		}
+	}
+
+	return activeNodeList
+}
+
+func runTestSuccessfulIncCounterIncreaseWithoutInstability(t *testing.T, clusterSize, numValidators, numRequests int) {
+	env := InitializeStabilityTest(t, numValidators, clusterSize)
+	env.sendRequests(numRequests, time.Millisecond*250)
+	waitUntil(t, env.chainEnv.counterEquals(int64(numRequests)), env.getActiveNodeList(), 120*time.Second, "incCounter matches expectation")
+}
+
+func TestSuccessfulIncCounterIncreaseWithoutInstability(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	t.Run("cluster=1,numValidators=1,req=35", func(t *testing.T) {
+		const clusterSize = 1
+		const numValidators = 1
+		const numRequests = 35
+
+		runTestSuccessfulIncCounterIncreaseWithoutInstability(t, clusterSize, numValidators, numRequests)
+	})
+
+	t.Run("cluster=5,numValidators=4,req=35", func(t *testing.T) {
+		const clusterSize = 5
+		const numValidators = 4
+		const numRequests = 35
+
+		runTestSuccessfulIncCounterIncreaseWithoutInstability(t, clusterSize, numValidators, numRequests)
+	})
+
+	t.Run("cluster=8,numValidators=6,req=35", func(t *testing.T) {
+		const clusterSize = 8
+		const numValidators = 6
+		const numRequests = 35
+
+		runTestSuccessfulIncCounterIncreaseWithoutInstability(t, clusterSize, numValidators, numRequests)
+	})
+}
+
+func runTestSuccessfulIncCounterIncreaseWithMildInstability(t *testing.T, clusterSize, numValidators, numBrokenNodes, numRequests int) {
+	env := InitializeStabilityTest(t, numValidators, clusterSize)
+	env.setSabotageValidators(numBrokenNodes)
+
+	wg := env.sabotageNodes(SabotageByKill, 4*time.Second, 1*time.Second)
+	env.sendRequests(numRequests, time.Millisecond*250)
+
+	wg.Wait()
```

Comment:
shouldn't we send the requests *after* the nodes are sabotaged? 😅 
```suggestion
	wg.Wait()
	env.sendRequests(numRequests, time.Millisecond*250)
```

Reply:
The idea is that the nodes break away *while* you send messages. That's why the node kills are in a separate thread.